pkg/core/native/native_test/common_test.go

```diff
@@ -117,20 +117,25 @@ func testGetSetCache(t *testing.T, c *neotest.ContractInvoker, name string, defa
 	}
 }
 
-func setNodesByRole(t *testing.T, designateInvoker *neotest.ContractInvoker, ok bool, r noderoles.Role, nodes keys.PublicKeys) {
+func setNodesByRole(t *testing.T, designateInvoker *neotest.ContractInvoker, ok bool, r noderoles.Role, nodes keys.PublicKeys, oldKeys ...keys.PublicKeys) {
 	pubs := make([]any, len(nodes))
 	for i := range nodes {
 		pubs[i] = nodes[i].Bytes()
 	}
 	if ok {
 		h := designateInvoker.Invoke(t, stackitem.Null{}, "designateAsRole", int64(r), pubs)
+		notification := stackitem.NewArray([]stackitem.Item{
+			stackitem.Make(int64(r)),
+			stackitem.Make(designateInvoker.Chain.BlockHeight()),
+		})
+		if len(oldKeys) > 0 {
+			notification.Append(stackitem.NewArray([]stackitem.Item{stackitem.NewByteArray(oldKeys[0][0].Bytes())}))
+			notification.Append(stackitem.NewArray([]stackitem.Item{stackitem.NewByteArray(nodes[0].Bytes())}))
+		}
```

Comment:
This way the test is not properly constructed. You need to take not only `[0]`-th element of nodes/oldKeys.